src/mokernels/moinput.jl

```diff
@@ -1,15 +1,44 @@
 """
-    MOInput(x::AbstractVector, out_dim::Integer)
+    IsotopicByFeatures(x::AbstractVector, out_dim::Integer)
 
-A data type to accomodate modelling multi-dimensional output data.
+`IsotopicByFeatures(x, out_dim)` has length `out_dim * length(x)`.
 
-`MOInput(x, out_dim)` has length `length(x) * out_dim`.
+```jldoctest
+julia> x = [1, 2, 3];
+
+julia> IsotopicByFeatures(x, 2)
+6-element IsotopicByFeatures{Int64, Vector{Int64}}:
+ (1, 1)
+ (1, 2)
+ (2, 1)
+ (2, 2)
+ (3, 1)
+ (3, 2)
+```
+
+Accommodates modelling multi-dimensional output data where all outputs are always observed.
+
+As shown above, an `IsotopicByFeatures` represents a vector of tuples.
+The first `out_dim` elements represent all outputs for the first input, the second
+`out_dim` elements represent the outputs for the second input, etc.
+
+See [Inputs for Multiple Outputs](@ref) in the docs for more info.
+"""
+struct IsotopicByFeatures{S,T<:AbstractVector{S}} <: AbstractVector{Tuple{S,Int}}
+    x::T
```

Comment:
How about
```suggestion
    xs::T
```
or even better
```suggestion
    features::T
```
to clarify what this is?

Reply:
If we make this change (which I am happy to do), I think this becomes more breaking, since I think the plan was to add an alias for `IsotopicByOutput` to `MOInput` (the old name) which uses `x` and `out_dim`.

Reply:
Ah, I thought it was going to be a breaking change anyways. In that case we can add it to the list for cleaning up struct field names more generally, in a separate PR

Reply:
Discussion moved to a new [issue](https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/issues/332)